src/geography.hpp

```diff
@@ -14,116 +20,148 @@ using S2GeographyIndexPtr = std::unique_ptr<s2geog::ShapeIndexGeography>;
 /*
 ** The registered Geography types
 */
-enum class GeographyType : std::int8_t { None = -1, Point, LineString, Polygon };
+enum class GeographyType : std::int8_t {
+    None = -1,
+    Point,
+    LineString,
+    Polygon,
+    MultiPoint,
+    MultiLineString,
+    MultiPolygon,
+    GeographyCollection
+};
 
 /*
 ** Thin wrapper around s2geography::Geography.
 **
-** Implements move semantics (avoid implicit copies).
-** Allows getting the geography type.
-** Wraps a s2geography::ShapeIndexGeography to speed-up operations
-** (the index is built on demand).
+** This wrapper implements the following specific features (that might
+** eventually move into s2geography::Geography?):
+**
+** - Implement move semantics only.
+** - add ``clone()`` method for explicit copy
+** - Add ``geog_type()`` method for getting the geography type
+** - Eagerly infer the geography type as well as other properties
+** - Encapsulate a lazy ``s2geography::ShapeIndexGeography`` accessible via ``geog_index()``.
 **
 */
 class Geography {
 public:
     Geography(const Geography&) = delete;
     Geography(Geography&& geog) : m_s2geog_ptr(std::move(geog.m_s2geog_ptr)) {
-        // std::cout << "Geography move constructor called: " << this <<
-        // std::endl;
+        extract_geog_properties();
```

Comment:
> Actually, I should have commented that on the case below (creating from s2geography), because for the line above in theory we could propagate the attributes from the passed Geography object?

Yes indeed we should simply propagate the attributes in this case.

In the case of creating from an s2geography object when we know the geography type, we could indeed provide the type explicitly instead of inferring it (and therefore avoid the inference overhead). We certainly want to avoid passing inconsistent s2geography object type vs. geography type, though, in order to prevent odd behavior or segfaults happening later. Checking for this consistency at runtime would be pretty similar to `extract_geog_properties` in terms of performance overhead I think. If we really want to optimize this, perhaps we could use template constructors + SFINAE or `if constexpr`...